pkg/lib/cockpit-components-typeahead-select.tsx

```diff
@@ -390,7 +401,7 @@ export const TypeaheadSelectBase: React.FunctionComponent<TypeaheadSelectProps>
         >
             <TextInputGroup isPlain>
                 <TextInputGroupMain
-          value={isFiltering ? filterValue : (selected?.content ?? '')}
+          value={isFiltering ? filterValue : (selected?.content ?? (props.selected || ''))}
```

Comment:
Can we make this palatable to upstream as well? Possibly with an extra flag to not break compatibility?

This would look a bit more symmetric with `selected?.content ?? props.selected ?? ''`

Reply:
I'll submit it. Thanks for the cleanup!

Reply:
Hmm, I am not so sure anymore about submitting it. This mixes "content" (user visible) and "value" (internal data structure stuff). This makes sense when content and value are the same, like with FileAutocomplete, but not in general... have to think about this.

Reply:
Nah, I count this as a hack to make FileAutocomplete work.

Reply:
Ack. Can this then go into the FileAutocomplete component somehow, instead of the lib component?

Reply:
I'll try. It might need larger surgery, maybe. We would need to make sure that "selected" is always included in "selectedOptions". And, hmm, this is actually necessary whenever isCreatable used... and also with the logs page where "selected" comes from the URL but the list of options is loaded asynchronously sometime later.

So I think it's more than just FileAutocomplete... back to the beginning.

Reply:
This changed to the more explicit `selectIsTrusted`, I think that's ok -- it's documented well.